src/audio/copier/copier.c

```diff
@@ -1453,22 +1474,17 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 		buffer_release(sink_c);
 	}
 
-	/* update the source format
-	 * used only for rare cases where two pipelines are connected by a shared
-	 * buffer and 2 copiers, this will set source format only for shared buffers
-	 * for a short time when the second pipeline already started
-	 * and the first one is not ready yet along with sink buffers params
+	/*
+	 * force update the source buffer format to cover cases where the source module
+	 * fails to set the sink buffer params
 	 */
 	if (!list_is_empty(&dev->bsource_list)) {
+		struct ipc4_audio_format *in_fmt;
 		source = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
 		source_c = buffer_acquire(source);
 
-		if (!source_c->hw_params_configured) {
```

Comment:
because some module does not update its sink buffer parameter, also does not set hw_params_configured, cause some CI test failed, so here force update buffer parameter.